integrations/test-gitlab-v2/client.py

```diff
@@ -0,0 +1,105 @@
+
+    async def _request(self, method: str, endpoint: str, **kwargs) -> dict:
+        url = f"{self.base_url}/{endpoint}"
+        logger.info(f"Making {method} request to {url}")
+        async with httpx.AsyncClient() as client:
+            response = await client.request(method, url, headers=self.headers, timeout=60, **kwargs)
+            logger.info(f"Response received from {url} with status code {response.status_code}")
+            response.raise_for_status()  # Raise error for non-2xx status
+            return response.json()
+
+    async def get_groups(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch groups from GitLab."""
+        endpoint = "groups"
+        logger.info("Fetching groups from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} groups")
+            yield page
+
+    async def get_merge_requests(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch merge requests from GitLab."""
+        endpoint = "merge_requests"
+        logger.info("Fetching merge requests from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} merge requests")
+            yield page
+
+    async def get_issues(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch issues from GitLab."""
+        endpoint = "issues"
+        logger.info("Fetching issues from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} issues")
+            yield page
+
+    async def get_projects(self) -> AsyncIterator[List[Dict[str, Any]]]:
+        """Fetch projects from GitLab."""
+        endpoint = "projects"
+        logger.info("Fetching projects from GitLab")
+        async for page in self._paginated_request("GET", endpoint):
+            logger.info(f"Fetched page with {len(page)} projects")
+            yield page
+
+    async def get_single_merge_request(self, mr_id: int) -> Dict[str, Any]:
+        """Fetch a single merge request by ID."""
+        endpoint = f"merge_requests/{mr_id}"
+        logger.info(f"Fetching merge request with ID {mr_id}")
+        merge_request = await self._request("GET", endpoint)
+        logger.info(f"Fetched merge request: {merge_request}")
+        return merge_request
+
+    async def get_single_issue(self, issue_id: int) -> Dict[str, Any]:
+        """Fetch a single issue by ID."""
+        endpoint = f"issues/{issue_id}"
+        logger.info(f"Fetching issue with ID {issue_id}")
+        issue = await self._request("GET", endpoint)
+        logger.info(f"Fetched issue: {issue}")
+        return issue
+
+    async def setup_webhook(self, app_host: str, webhook_token: str) -> None:
+        """Setup GitLab webhook for a project."""
+        project_id = ocean.integration_config.get("project_id")
```

Comment:
we have to use group hooks instead of creating the webhook for a single project.